src/web.rs

```diff
@@ -48,3 +51,145 @@ impl WebWindowHandle {
         Self { id }
     }
 }
+
+/// Raw window handle for a Web canvas registered via [`wasm-bindgen`].
+///
+/// [`wasm-bindgen`]: https://crates.io/crates/wasm-bindgen
+#[non_exhaustive]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct Wbg02CanvasWindowHandle {
+    /// The pointer to the [`JsValue`] of an [`HtmlCanvasElement`].
+    ///
+    /// [`HtmlCanvasElement`]: https://docs.rs/web-sys/latest/web_sys/struct.HtmlCanvasElement.html
+    /// [`wasm-bindgen`]: https://crates.io/crates/wasm-bindgen
+    /// [`JsValue`]: https://docs.rs/wasm-bindgen/latest/wasm_bindgen/struct.JsValue.html
+    pub obj: NonNull<c_void>,
+}
+
+impl Wbg02CanvasWindowHandle {
+    /// Create a new handle to a pointer to [`HtmlCanvasElement`].
+    ///
+    /// [`HtmlCanvasElement`]: https://docs.rs/web-sys/latest/web_sys/struct.HtmlCanvasElement.html
+    ///
+    /// ## Example
+    /// ```no_run
+    /// # use raw_window_handle::Wbg02CanvasWindowHandle;
+    /// # use core::{ffi::c_void, ptr::NonNull};
+    /// # fn get_canvas() -> NonNull<c_void> { unimplemented!() }
+    /// let obj: NonNull<c_void> = get_canvas();
+    /// let mut window_handle = Wbg02CanvasWindowHandle::new(obj);
+    /// /* set fields */
+    /// ```
```

Comment:
Nit: No fields to set later, make it actually explain a little bit about what to do, and let CI run it.

```suggestion
    /// # Example
    ///
    /// ```
    /// # use core::ffi::c_void;
    /// # use core::ptr::NonNull;
    /// # use raw_window_handle::Wbg02CanvasWindowHandle;
    /// # type HtmlCanvasElement = ();
    /// # type JsValue = ();
    /// let canvas: &HtmlCanvasElement;
    /// # canvas = &();
    /// let value: &JsValue = &canvas; // Deref to `JsValue`
    /// let obj: NonNull<c_void> = NonNull::from(value).cast();
    /// let mut handle = Wbg02CanvasWindowHandle::new(obj);
    /// ```
```